sklearn/metrics/_pairwise_distances_reduction/_gemm_term_computer.pxd.tp

```diff
@@ -20,7 +20,7 @@ from libcpp.vector cimport vector
 
 {{for name_suffix, upcast_to_float64, INPUT_DTYPE_t, INPUT_DTYPE in implementation_specific_values}}
 
-cdef class GEMMTermComputer{{name_suffix}}:
+cdef class MiddleTermComputer{{name_suffix}}:
     cdef:
         const {{INPUT_DTYPE_t}}[:, ::1] X
         const {{INPUT_DTYPE_t}}[:, ::1] Y
```

Comment:
Those two attributes and (`{X,Y}_c_upcast` with their conditional Tempita definition) have to be pulled down in `DenseDenseMiddleTermComputer`.

Attributes for `SparseSparseMiddleTermComputer` must be the CSR matrices internal arrays, like the ones of `SparseSparseDatasetsPairs`:

https://github.com/scikit-learn/scikit-learn/blob/df626e43ac3c99da2aeb005709d77395bd717e6a/sklearn/metrics/_pairwise_distances_reduction/_datasets_pair.pxd.tp#L44-L51